### PR TITLE
Implement AbstractLoggingException

### PR DESCRIPTION
**Pull request by doncem**
Since [v1.5.0](https://github.com/hmcts/java-logging/releases/tag/1.5.0) every error log must have alert level and error code. Both can be disabled via [environment variables](https://github.com/hmcts/java-logging#additional-logback-configuration)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/logging/ErrorCode.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package uk.gov.hmcts.reform.sendletter.logging;
+
+@SuppressWarnings({"squid:S1118", "HideUtilityClassConstructor"})
+public final class ErrorCode {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/sendletter/logging/ErrorCode.java]: This is going to be repeated every time I guess.Best to extract in logging project itself.May be a separate PR.

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/logging/ErrorCode.java]: yes. should definitely speak about it tomorrow or next tuesday

**Comment by doncem** [comment]
Will update with new `java-logging` version
